### PR TITLE
show icon once image added to collection

### PR DESCRIPTION
**Pull request by tsop14**
User sees little folder icon (or whatever imagery we use in the future) once an image has been added to a new collection. 

adding-images

### CODE REVIEW

File: kahuna/public/js/services/api/collections-api.js [hunk 1]
```diff
@@ -60,13 +60,37 @@ apiServices.factory('collections',
...
     }
 
+    function getCollectionsX(apiImageResource) {
+        return apiImageResource.data.collections.map(e => e.data.pathId);
+    }
+
+    function untilNewCollectionAppears(image, collectionAdded) {
+        return image.get().then( (apiImage) => {
+            const apiCollections = getCollectionsX(apiImage);
+            if(apiCollections.indexOf(collectionAdded.data.pathId) > -1) {
```
Comment (theefer) [inline kahuna/public/js/services/api/collections-api.js]: Nitpick mode: space after `if`.

Also: can we have `Array.prototype.include` already ;_;

File: kahuna/public/js/services/api/collections-api.js [hunk 2]
```diff
@@ -60,13 +60,37 @@ apiServices.factory('collections',
...
+                $rootScope.$emit('image-updated', newImage, image);
+                return newImage;
+            })
+
         ).toJS();
 
         return $q.all(promises);
     }
 
+    function getCollectionsX(apiImageResource) {
```
Comment (theefer) [inline kahuna/public/js/services/api/collections-api.js]: Is the name deliberate?

**Comment by theefer** [comment]
Minor comments, functionally :+1:

File: kahuna/public/js/services/api/collections-api.js (hunk 1 above)
Comment (jamesgorrie) [inline kahuna/public/js/services/api/collections-api.js]: :boom:Just tried it in chrome console, it seems to work:boom:

Comment (theefer) [inline kahuna/public/js/services/api/collections-api.js]: Maybe a bit too [edgy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes#Browser_compatibility), esp as Traceur doesn't seem to translate it...?

File: kahuna/public/js/services/api/collections-api.js [hunk 3]
```diff
@@ -60,13 +60,37 @@ apiServices.factory('collections',
...
     function addImagesToCollection(images, path) {
+
         const promises = images.map(image =>
             image.perform('add-collection', {body: {data: path}})
+            .then(collectionAdded => apiPoll(() =>
+                 untilNewCollectionAppears(image, collectionAdded)
+            ))
+            .then(newImage => {
+                $rootScope.$emit('image-updated', newImage, image);
+                return newImage;
```
Comment (akash1810) [inline kahuna/public/js/services/api/collections-api.js]: is the returned value used?

**Comment by theefer** [comment]
Is this little icon still the plan, vs the coloured leaf labels discussed yesterday?

**Comment by tsop14** [comment]
> Is this little icon still the plan, vs the coloured leaf labels discussed yesterday?

Little icon is on the out but it wasn't created in this PR. Going to change to coloured leaf in another PR soon.

File: kahuna/public/js/services/api/collections-api.js (hunk 3 above)
Comment (theefer) [inline kahuna/public/js/services/api/collections-api.js]: Makes sense not to swallow the value unexpectedly imho.

**Comment by theefer** [comment]
OK cool!

**Pull request merged**